ihatemoney/tests/budget_test.py

```diff
@@ -982,6 +987,85 @@ def test_statistics(self):
         self.assertRegex(response.data.decode("utf-8"), re.compile(regex1, re.DOTALL))
         self.assertRegex(response.data.decode("utf-8"), re.compile(regex2, re.DOTALL))
 
+        # Check that the "monthly expenses" table has a single month and the correct amount
+        regex = r"<tbody>\s*<tr>\s*<td>August 2011</td>\s*<td>\$40.00</td>\s*</tr>\s*</tbody>"
+        self.assertRegex(response.data.decode("utf-8"), regex)
+
+        # Add bills for other months and check monthly expenses again
+        self.client.post(
+            "/raclette/add",
+            data={
+                "date": "2011-12-31",
+                "what": "champomy",
+                "payer": 1,
+                "payed_for": [1, 2],
+                "amount": "10",
+            },
+        )
+        self.client.post(
+            "/raclette/add",
+            data={
+                "date": "2011-08-01",
+                "what": "ice cream",
+                "payer": 2,
+                "payed_for": [1, 2],
+                "amount": "10",
+            },
+        )
+        months = [
+            ("December 2011", r"\$10.00"),
+            ("November 2011", r"\$0.00"),
+            ("October 2011", r"\$0.00"),
+            ("September 2011", r"\$0.00"),
+            ("August 2011", r"\$50.00"),
+        ]
+        months_fmt = [
+            r"<tr>\s*<td>{}</td>\s*<td>{}</td>\s*</tr>".format(month, amount)
+            for month, amount in months
+        ]
```

Comment:
Actually, I find it kinda weird to check the output of the HTML like this, because it means that if we change the format of the table we also need to change the tests, even though we're not changing the actual behavior of the code.

I wonder if we might instead create a model function, and test that it works properly with python-related tests (more unit tests than functional tests, though). We might also want to have this information on the `/api` endpoint.

What do you think?

Reply:
As explained below, I just implement this new testing approach!

For the API, I'm not sure. The client could just request the list of bills and compute any statistics itself.